File: utils/flavor.js
import { readFlavor } from './firebase.js';
import logger from './logger.js';

export async function pickLine(key) {
  try {
    const all = await readFlavor();
    const lines = all[key];
    if (Array.isArray(lines) && lines.length > 0) {
      return lines[Math.floor(Math.random() * lines.length)];
    }
  } catch (err) {
    logger.error(`Failed to read flavor key [${key}]:`, err);
  }
  return '';
}

File: internal/flavor/flavor.go
package flavor

import (
	"context"
	"log/slog"
	"math/rand/v2"
)

// Reader can fetch flavor lines from a backing store.
type Reader interface {
	ReadFlavor(ctx context.Context) (map[string][]string, error)
}

// PickLine returns a random line for the given flavor key.
// Returns empty string if the key doesn't exist or the store is unavailable.
func PickLine(ctx context.Context, r Reader, key string) string {
	all, err := r.ReadFlavor(ctx)
	if err != nil {
		slog.Error("failed to read flavor", "key", key, "err", err)
		return ""
	}
	lines := all[key]
	if len(lines) == 0 {
		return ""
	}
	return lines[rand.IntN(len(lines))]
}
